packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.scss

```diff
@@ -153,64 +148,105 @@
   align-items: center;
 }
 
-.actionsOneline {
-  position: relative;
+.innerTextPadding {
+  span, .innerText > span {
+    @include padding-right(4px); // Add padding between text and hyperlink.
+  }
 }
 
 .dismissal {
   @include focus-border();
   min-width: 0;
-}
-.dismissalOneline .dismissal {
-  @include margin-right(-8px);
+  position: absolute;
+  top: 0;
+  @include right(0);
 }
 
-// Add space between adjacent MessageBars.
-.root + .root {
-  margin-top: 6px;
-}
+.singleline {
+  height: 32px;
 
-.innerTextPadding {
-  @include padding-right(24px); // Add padding if we have a dismiss to prevent button overlapping text.
+  .dismissalOneline {
+    min-width: 16px;
+    min-height: 16px;
+    @include padding(8px, 8px, 0px, 12px);
 
-  span, .innerText > span {
-    @include padding-right(4px); // Add padding between text and hyperlink.
+    .innterTextPadding {
+      @include padding-right(48px);
+    }
   }
-}
-
-// When we can have multiple lines in the message bar.
-.multiline > .content > .actionables {
-  flex-direction: column;
-}
-
-// When we should have only a single line of text in the message bar.
-.singleline {
   .content {
-    .icon{
-      align-items: center;
+    align-items: center;
+    padding: 8px 0px;
+    .icon {
+      padding: 0px 8px;
     }
 
-    .actionables > .text {
-      justify-content: center;
+    .text {
       align-items: center;
 
       .innerText,
       .innerTextPadding {
         max-height: 1.3em;
-        line-height: 1.3em;
         overflow-x: hidden;
         text-overflow: ellipsis;
         white-space: nowrap;
       }
     }
   }
 
-  .content > .actionables {
-    flex-direction: row;
+  .actionsOneline {
+    position: relative;
   }
-}
 
-// Because of an odd behaviour in other CSS, ms-MessageBar--remove causes children's icons to use 8px, and not inherit.
-.root :global(.ms-Icon--Cancel) {
-  font-size: 14px;
+  // Override one line values when we have buttons / actionable items
+  &.actionableOneline {
+    height: 48px;
+
+    .dismissalOneline {
```

Comment:
As discussed.
https://css-tricks.com/specifics-on-css-specificity/

Being this is a library, if we're 4 levels deep, consumers will have to match or add !important. Neither of which is a great experience. 

Updated.